lib/tests/formulae_dependents.rb

```diff
@@ -166,6 +166,23 @@ def install_dependent(dependent, testable_dependents, args:, build_from_source:
           test "brew", "link", dependent.full_name
         end
         test "brew", "install", "--only-dependencies", dependent.full_name
+
+        dependent.deps.each do |dependency|
+          dependency_f = dependency.to_formula
+
+          # Uninstall build- or test-only dependencies to avoid `brew linkage --test`
```

Comment:
```suggestion
          # Uninstall build-only or test-only dependencies to avoid `brew linkage --test`
```

Reply:
Also, we already do uninstall this for non-dependents testing, perhaps we could/should use the logic (or adjust the ordering?)

Reply:
Yea, this step smells reusable to me. I just haven't worked out the right way to refactor things. I'll take a closer look.